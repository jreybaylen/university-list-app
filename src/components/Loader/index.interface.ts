interface LoaderProps {
    show: boolean
}

export type {
    LoaderProps
}