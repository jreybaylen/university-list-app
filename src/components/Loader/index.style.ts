import { StypesProps } from '@interface/style.interface'

const styles: StypesProps = {
    container: {
        padding: 20,
        width: '100%',
        margin: 'auto',
        maxWidth: 1000,
        fontWeight: 500,
        fontSize: '1.2rem',
        textAlign: 'center'
    }
}

export {
    styles
}