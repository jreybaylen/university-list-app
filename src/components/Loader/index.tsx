import { Fragment } from 'react'

import { styles } from './index.style'
import { LoaderProps } from './index.interface'

function Loader (props: LoaderProps): JSX.Element {
    if (!Boolean(props.show)) {
        return <Fragment />
    }

    const loaderElement = (
        <div style={ styles.container }>Loading data...</div>
    )

    return loaderElement
}

export {
    Loader
}