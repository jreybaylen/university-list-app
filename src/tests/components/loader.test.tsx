import { render } from '@testing-library/react'

import { Loader } from '@components/Loader'

describe('<Loader /> Component', () => {
    it('Should render without crashing', () => {
        const wrapper = render(<Loader show={ true } />)

        expect(wrapper).toMatchSnapshot()
    })

    it('Should render component when "show" prop is "true"', () => {
        const { getByText } = render(<Loader show={ true } />)

        expect(getByText(/loading data.../i)).toBeInTheDocument()
    })

    it('Should not render component when "show" prop is "false"', () => {
        const { queryByText } = render(<Loader show={ false } />)

        expect(queryByText(/loading data.../i)).toBe(null)
    })
})