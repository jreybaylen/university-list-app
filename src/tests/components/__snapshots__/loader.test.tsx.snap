// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> Component Should render without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="padding: 20px; width: 100%; margin: auto; max-width: 1000px; font-weight: 500; font-size: 1.2rem; text-align: center;"
      >
        Loading data...
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="padding: 20px; width: 100%; margin: auto; max-width: 1000px; font-weight: 500; font-size: 1.2rem; text-align: center;"
    >
      Loading data...
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
